internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -214,6 +217,82 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 			nextCheckTimes = append(nextCheckTimes, nextCheck)
 		}
 	}
+
+	// check machine conditions
+	for _, c := range t.MHC.Spec.Checks.UnhealthyMachineConditions {
+		machineCondition := getMachineCondition(t.Machine, c.Type)
+
+		// Skip when current machine condition is different from the one reported
+		// in the MachineHealthCheck.
+		if machineCondition == nil || machineCondition.Status != c.Status {
+			continue
+		}
+
+		// If the machine condition has been in the unhealthy state for longer than the
+		// timeout, mark as unhealthy and collect the message.
+		timeoutSecondsDuration := time.Duration(ptr.Deref(c.TimeoutSeconds, 0)) * time.Second
+
+		if machineCondition.LastTransitionTime.Add(timeoutSecondsDuration).Before(now) {
+			foundUnhealthyCondition = true
+			unhealthyMessages = append(unhealthyMessages, fmt.Sprintf("Machine condition %s is %s for more than %s", c.Type, c.Status, timeoutSecondsDuration.String()))
+			unhealthyReasons = append(unhealthyReasons, "UnhealthyMachine")
+			logger.V(3).Info("Target is unhealthy: machine condition is in state longer than allowed timeout", "condition", c.Type, "state", c.Status, "timeout", timeoutSecondsDuration.String())
+			continue
+		}
+
+		durationUnhealthy := now.Sub(machineCondition.LastTransitionTime.Time)
+		nextCheck := timeoutSecondsDuration - durationUnhealthy + time.Second
+		if nextCheck > 0 {
+			nextCheckTimes = append(nextCheckTimes, nextCheck)
+		}
+	}
+
+	// If any unhealthy conditions were found, set the combined status
+	if foundUnhealthyCondition {
+		// Determine the primary reason based on a consistent priority order:
+		// 1. If both node and machine conditions are present, use a combined reason
+		// 2. Otherwise use the specific reason for the type that failed
+		var primaryReason, v1beta1Reason string
+		if len(unhealthyReasons) > 0 {
+			// Check if we have both node and machine reasons
+			hasNodeReason := false
+			hasMachineReason := false
+			for _, reason := range unhealthyReasons {
+				switch reason {
+				case "UnhealthyNode":
+					hasNodeReason = true
+				case "UnhealthyMachine":
+					hasMachineReason = true
+				}
+			}
+
+			if hasNodeReason && hasMachineReason {
+				// Both types of conditions are unhealthy - use machine reason but indicate it's combined
+				primaryReason = clusterv1.MachineHealthCheckUnhealthyMachineReason
+				v1beta1Reason = clusterv1.UnhealthyMachineConditionV1Beta1Reason
+			} else if hasMachineReason {
+				primaryReason = clusterv1.MachineHealthCheckUnhealthyMachineReason
+				v1beta1Reason = clusterv1.UnhealthyMachineConditionV1Beta1Reason
+			} else if hasNodeReason {
+				primaryReason = clusterv1.MachineHealthCheckUnhealthyNodeReason
+				v1beta1Reason = clusterv1.UnhealthyNodeConditionV1Beta1Reason
+			}
+		}
```

Comment:
I would simplify this to


```suggestion
		reason := clusterv1.MachineHealthCheckUnhealthyNodeReason
		v1beta1Reason := clusterv1.UnhealthyNodeConditionV1Beta1Reason
		if len(unhealthyMachineMessages) > 0 {
			reason = clusterv1.MachineHealthCheckUnhealthyMachineReason
			v1beta1Reason = clusterv1.UnhealthyMachineConditionV1Beta1Reason
		}
```